### PR TITLE
[BugFix] Cancel fragment instance not query when close external scanner

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

If cancel query, other fragment instances (scanners) will be cancelled, and spark connector will report SocketTimeoutException when get next data from BE.

Query executed by external connectors does not need to report exec state
to FE.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/runtime/external_scan_context_mgr.cpp [hunk 1]
```diff
@@ -107,19 +107,25 @@ Status ExternalScanContextMgr::clear_scan_context(const std::string& context_id)
...
                 _active_contexts.erase(context_id);
-                Status::OK();
+                return Status::OK();
             }
             iter = _active_contexts.erase(iter);
         }
     }
     if (context != nullptr) {
         // cancel pipeline
+        auto fragment_instance_id = context->fragment_instance_id;
```
Comment (silverbullet233) [inline be/src/runtime/external_scan_context_mgr.cpp:117]:
```suggestion
        const auto& fragment_instance_id = context->fragment_instance_id;
```

Comment (wyb) [reply be/src/runtime/external_scan_context_mgr.cpp]: done

**Review by silverbullet233 — APPROVED** [review]
LGTM

**Review by satanson — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**